lib/googleauth/signet.rb

```diff
@@ -77,6 +77,22 @@ def notify_refresh_listeners
           block.call(self)
         end
       end
+
+      def retry_with_error(retry_count = 5)
+        begin
+          yield
+        rescue => e
+          raise e if e.is_a?(Signet::AuthorizationError) || e.is_a?(Signet::ParseError)
+
+          if retry_count > 0
+            retry_count -= 1
+            retry
```

Comment:
@blowmage 
Thank you for your review and comment!

> Do you want to implement an incremental backoff before the retry?

Yes. 
I think there is no problem. 

I think this rescue block behaves like the following:
- `retry_count == 5` =>retry
- `retry_count == 4` =>retry
- `retry_count == 3` =>retry
- `retry_count == 2` =>retry
- `retry_count == 1` =>retry
- `retry_count == 0` =>raise

example:

```ruby
client = GCECredentials.new
client.retry_with_error do
  puts 'hey'
  raise
end
```

```
hey # first execute
hey # 1st retry
hey # 2nd retry
hey # 3rd retry
hey # 4th retry
hey # 5th retry
# raise Signet::AuthorizationError
```

Do you have something to worry about this behavior?